core/src/apps/ripple/sign_tx.py

```
@@ -24,15 +26,19 @@ async def sign_tx(msg: RippleSignTx, keychain: Keychain) -> RippleSignedTx:
     await paths.validate_path(keychain, msg.address_n)
 
     node = keychain.derive(msg.address_n)
     source_address = helpers.address_from_public_key(node.public_key())
 
+    IS_PAYMENT_TRANSACTION = helpers.is_payment_transaction(msg)
+
+    if IS_PAYMENT_TRANSACTION:
+        payment = msg.payment  # local_cache_attribute
+
+        if payment.amount > helpers.MAX_ALLOWED_AMOUNT:
+            raise ProcessError("Amount exceeds maximum allowed amount.")
```

Ruff (0.14.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)